frontend/src/components/ui/textarea.tsx

```diff
@@ -34,4 +35,58 @@ const Textarea = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
 );
 Textarea.displayName = "Textarea";
 
+export const DebouncedTextarea = React.forwardRef<
+  HTMLTextAreaElement,
+  TextareaProps & {
+    value: string;
+    onValueChange: (value: string) => void;
+    delay?: number;
+  }
+>(({ className, onValueChange, ...props }, ref) => {
+  const { value, onChange } = useDebounceControlledState<string>({
+    initialValue: props.value,
+    delay: props.delay,
+    onChange: onValueChange,
+  });
+
+  return (
+    <Textarea
+      ref={ref}
+      className={className}
+      {...props}
+      onChange={(evt) => onChange(evt.target.value)}
+      value={value}
+    />
+  );
+});
+DebouncedTextarea.displayName = "DebouncedTextarea";
+
+export const OnBlurredTextarea = React.forwardRef<
+  HTMLTextAreaElement,
+  TextareaProps & {
+    value: string;
+    onValueChange: (value: string) => void;
+  }
+>(({ className, onValueChange, ...props }, ref) => {
+  const [internalValue, setInternalValue] = React.useState(props.value);
+  return (
+    <Textarea
+      ref={ref}
+      className={className}
+      {...props}
+      onChange={(event) => setInternalValue(event.target.value)}
+      onBlur={() => onValueChange(internalValue)}
+      onKeyDown={(event) => {
+        if (!(event.ctrlKey && event.key === "Enter")) {
+          return;
+        }
+        event.preventDefault();
+        onValueChange(internalValue);
+      }}
+      value={internalValue}
+    />
+  );
```

Comment:
never nice!

Reply:
😮

Reply:
:open_mouth:

Reply:
Am I misunderstanding the role of `internalValue` here? 😆